rpc/core/env.go

```diff
@@ -100,6 +107,69 @@ type Environment struct {
 	genChunks []string
 }
 
+// InitGenesisChunks checks whether it makes sense to create a cache of chunked
+// genesis data. It is called on Node startup and should be called only once.
+// Rules of chunking:
+// - if the genesis file's size is <= 16MB, then no chunking. An `Environment`
+// object will store a pointer to the genesis in its GenDoc field. Its genChunks
+// field will be set to nil. `/genesis` RPC API will return the GenesisDoc itself.
+// - if the genesis file's size is > 16MB, then use chunking. An `Environment`
+// object will store a slice of base64-encoded chunks in its genChunks field. Its
+// GenDoc field will be set to nil. `/genesis` RPC API will redirect users to use
+// the `/genesis_chunked` API.
+func (env *Environment) InitGenesisChunks() error {
+	if len(env.genChunks) > 0 {
+		// we already computed the chunks, return.
+		return nil
+	}
+
+	if env.GenDoc == nil {
+		// chunks not computed yet, but no genesis available.
+		// This should not happen.
+		return errors.New("pointer to genesis is nil")
+	}
+
+	data, err := cmtjson.Marshal(env.GenDoc)
+	if err != nil {
+		return err
+	}
+
+	// If genesis is less than 16MB, then no chunking.
+	// Keep a pointer to a GenesisDoc in env.GenDoc.
+	if len(data) <= genesisChunkSize {
+		env.genChunks = nil
```

Comment:
But the `genesis_chunked` endpoint still remains active and returns the chunks ? So we either edit that to create a on-demand chunk, or we do not delete the chunks here. I think , for now, it is simpler to just not set it to nil. That is, if we have only one chunk (<16MB) , then its not that big of an overhead if we keep it in memory.

Reply:
The `genesis_chunked` endpoint remains active but returns an error.
See [here](https://github.com/cometbft/cometbft/blob/6bfe1e15e06ca3581fdb392adddb91e9a1ca7b68/rpc/core/net.go#L121). I changed the endpoint so that:
- if there are no chunks but the request's chunk id = 0, then it returns the genesis serialized to JSON. This preserves the current behavior of `/genesis_chunked`
- if there are no chunks and the requested chunk id >= 1, it returns an error. This is also keeps the existing behavior.

This way, we can ditch the chunk's slice.

Reply:
Ah I missed that for some reason even though I searched for it . That's ok then. I guess you are just delaying the chunk creation until it is actually needed and then it is created only once.